### PR TITLE
Implement multipart/form-data

### PR DESCRIPTION
**Pull request by cgay**
Implement multipart/form-data. This isn't complete but is enough to use in play.opendylan.org. 

Also added get-query-values to retrieve all URL/form values under the same key.

### CODE REVIEW

File: server/core/request.dylan [hunk 1]
```diff
@@ -327,81 +356,41 @@ define inline function empty-line?
...
+              val
+            end;
+  if (as-type & instance?(val, <string>))
     as(as-type, val)
   else
     val
   end
 end function get-query-value;
 
+define inline function get-query-values (key :: <string>) => (values :: <list>)
```
Comment (housel) [inline server/core/request.dylan]: Why not generic `<sequence>` for the API?

Comment (cgay) [reply server/core/request.dylan]: Done, thanks.

File: documentation/source/reference/server.rst [hunk 2]
```diff
@@ -342,6 +342,13 @@ The HTTP-SERVER module
    :parameter #key as: An instance of ``false-or(<type>)``.
    :value value: An instance of ``<object>``.
 
+.. function:: get-query-values
+
+   :signature: get-query-values (key) => (values)
+
+   :parameter key: An instance of ``<string>``.
+   :value values: An instance of ``<list>``.
```
Comment (housel) [inline documentation/source/reference/server.rst]: Need to update this to match

Comment (cgay) [reply documentation/source/reference/server.rst]: done

**Pull request merged**